runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/Timer.java

```diff
@@ -53,6 +53,12 @@ public Timer(long period, ChronoUnit unit, BufferedImage image, Plugin plugin)
 		endTime = startTime.plus(duration);
 	}
 
+	public void reset()
```

Comment:
If you change the fields to `protected` access level here, you can move this method into BarrowsPrayerDrainTimer as a private instance method instead of adding it here.

Reply:
I'll leave my comment from discord here. I'm not sure time should be mutable like this but I don't have a strong opinion.

Reply:
I would much prefer just recreating a new timer.

Reply:
Or just dont use timer at all, because it do not fits this scenario

Reply:
How would you avoid a concurrent modification exception? How could you hook into the timer expiration to create a new one?